starter_code/src/movies.js

```diff
@@ -2,14 +2,106 @@
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
 
+//                                                                   Is it not possible to use .map here?
+function orderByYear(el) {
+    const movYearOrd = el.slice().sort(function (a, b) {
+        if (a.year === b.year) {
+            return a.title.localeCompare(b.title);
+        }
+        return a.year - b.year;
+    });
+    if (el.length <= 0) {
+        return null;
+    }
+    return movYearOrd;
+}
+console.log(orderByYear(movies));
+
+
+
+
+
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(el) {
+    const splbrgDrmMvs = el.filter(function (el) {
+        if (el.director === "Steven Spielberg" &&
+            el.genre.includes("Drama")) {
+            return true;
+        }
+    })
+    return splbrgDrmMvs.length;
+}
+
+console.log(howManyMovies(movies));
+
+
+
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
+function orderAlphabetically(el) {
+    const alphOrd =
+        el.slice().sort(function (a, b) {
+            return a.title.localeCompare(b.title);
+
+        }).map(function (el) {
+            return el.title
+        }).slice(0, 20)
+    return alphOrd;
+}
+console.log(alphOrd(movies));
+
+
+
 
 // Iteration 4: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(el) {
+    if (!el.length) return 0;
+    let avgAllRts = 0;
+    const sumAllRts = el.reduce(function (acc, rateVal) {
+        if (!rateVal.rate) return acc;
+        return acc + rateVal.rate;
+    }, 0);
+    avgAllRts = parseFloat((sumAllRts / el.length).toFixed(2));
+    return avgAllRts;
+}
+console.log(avgAllRts(movies));
+
+
+
+
 // Iteration 5: Drama movies - Get the average of Drama Movies
+// Using ratesAverage function from above :)
+
+function dramaMoviesRate(el) {
+    const dramaMovies = el.filter(function (el) {
+        if (el.genre.includes("Drama")) {
+            return true;
+        }
+    });
+    return ratesAverage(dramaMovies);
+}
+console.log(dramaMoviesRate(movies));
+
+
+
 
 // Iteration 6: Time Format - Turn duration of the movies from hours to minutes
+function turnHoursToMinutes(el) {
+    const moviesInMin = el.map(function (el) {
```

Comment:
Hint: you would need to split at the empty space, and turn into a number. (+ replace the string "min" and "h" with empty strings)

Then you would have to multiply the hours by 60 to get the minutes and add that to the minutes you already have. 
Create a shallow copy and replace the old duration with the new one. 

There are many ways to do it of course, this is one of them only :). It's tricky but I'm sure you can figure it out! 💯

Reply:
Thanks for detailed hint! we worked on it again in class today and it is clear now, just need to practice on a couple of different examples.
Vielen Dank :)